spring-cloud-stream-metrics/src/main/java/org/springframework/cloud/stream/metrics/config/MetricJsonSerializer.java

```diff
@@ -40,7 +40,12 @@
 @JsonComponent
 public class MetricJsonSerializer {
 
-	final static DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+	final static ThreadLocal<DateFormat> df = new ThreadLocal<DateFormat>(){
+			@Override
+			protected DateFormat initialValue() {
+					return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+			}
+	};
```

Comment:
I am not a fan of leaving dangling `ThreadLocals` attached to arbitrary threads. In [Spring Integration we usually use a dynamic pool of formatters](https://github.com/spring-projects/spring-integration/blob/4.3.x/spring-integration-core/src/main/java/org/springframework/integration/transformer/SyslogToMapTransformer.java#L93).

That said, it depends on how long this will live; if we're going to Java8 soon, we can let this stand and move to `DateTimeFormatter` which is thread-safe. @mbogoevici ??